Error during tag/build when using testing-library/jest-dom matchers

<!--

Before submitting a new issue, please search past issues (open or closed).  

-->

### Describe the bug

I receive the following error when running `bbit tag` or `bbit build`

```
✖ /Users/someuser/Library/Caches/Bit/capsules/0b03fbda727426b9600433bee84d17eff31d0b2a/my-org.custom-environment-test_controls_fancy-input@0.0.1/FancyInput.test.tsx:9:19 - error TS2339: Property 'toHaveValue' does not exist on type 'JestMatchersShape<Matchers<void, HTMLElement>, Matchers<Promise<void>, HTMLElement>>'.
9     expect(input).toHaveValue("test value");
```

### Steps to Reproduce

1. Open this project:
[custom-env-test.zip](https://github.com/teambit/bit/files/6211772/custom-env-test.zip)
2. Run the following commands:
```
yarn
bbit link
bbit build
```

### Expected Behavior

`bbit build` should run successfully

### Specifications

- Bit version: 0.0.331 (@teambit/legacy: 1.0.42)
- Node version: v14.15.5
- yarn version: 1.22.10
- Platform: macOS 11.2.3

### Additional context

I have found a workaround for this problem. Here it is:
1. Add this line `import '@testing-library/jest-dom';` at the top of `FancyInput.test.tsx`.
2. Change package manager to `yarn` in `workspace.jsonc`.

Now `bbit build` runs successfully.

--- a/scopes/react/react/react.env.ts
+++ b/scopes/react/react/react.env.ts
@@ -259,6 +259,7 @@ export class ReactEnv implements Environment {
         // '@types/react-router-dom': '^5.0.0', // TODO - should not be here (!)
         // This is added as dev dep since our jest file transformer uses babel plugins that require this to be installed
         '@babel/runtime': '7.12.18',
+        '@types/testing-library__jest-dom': '5.9.5',
       },
       // TODO: take version from config
       peerDependencies: {

